*Dataset URL:*  https://www.kaggle.com/datasets/asaniczka/amazon-products-dataset-2023-1-4m-products

In [1]:
import pandas as pd
import numpy as np

In [2]:
df_category = pd.read_csv(r"C:\Users\Akhand Pratap Singh\Desktop\archive1\amazon_categories.csv", )
df_product = pd.read_csv(r"C:\Users\Akhand Pratap Singh\Desktop\archive1\amazon_products.csv")

In [3]:
df_category.head(5)

id                     category_name
0   1          Beading & Jewelry Making
1   2                 Fabric Decorating
2   3       Knitting & Crochet Supplies
3   4              Printmaking Supplies
4   5  Scrapbooking & Stamping Supplies

In [4]:
df_product.head(5)

asin                                              title  \
0  B014TMV5YE  Sion Softside Expandable Roller Luggage, Black...   
1  B07GDLCQXV  Luggage Sets Expandable PC+ABS Durable Suitcas...   
2  B07XSCCZYG  Platinum Elite Softside Expandable Checked Lug...   
3  B08MVFKGJM  Freeform Hardside Expandable with Double Spinn...   
4  B01DJLKZBA  Winfield 2 Hardside Expandable Luggage with Sp...   

                                              imgUrl  \
0  https://m.media-amazon.com/images/I/815dLQKYIY...   
1  https://m.media-amazon.com/images/I/81bQlm7vf6...   
2  https://m.media-amazon.com/images/I/71EA35zvJB...   
3  https://m.media-amazon.com/images/I/91k6NYLQyI...   
4  https://m.media-amazon.com/images/I/61NJoaZcP9...   

                             productURL  stars  reviews   price  listPrice  \
0  https://www.amazon.com/dp/B014TMV5YE    4.5        0  139.99       0.00   
1  https://www.amazon.com/dp/B07GDLCQXV    4.5        0  169.99     209.99   
2  https://www.amazon.com/dp/B07XSCCZYG    4.6        0  365.49     429.99   
3  https://www.amazon.com/dp/B08MVFKGJM    4.6        0  291.59     354.37   
4  https://www.amazon.com/dp/B01DJLKZBA    4.5        0  174.99     309.99   

   category_id  isBestSeller  boughtInLastMonth  
0          104         False               2000  
1          104         False               1000  
2          104         False                300  
3          104         False                400  
4          104         False                400

In [5]:
df = pd.merge(df_product, df_category, left_on= 'category_id', right_on = 'id', how = 'left')
df.head()

asin                                              title  \
0  B014TMV5YE  Sion Softside Expandable Roller Luggage, Black...   
1  B07GDLCQXV  Luggage Sets Expandable PC+ABS Durable Suitcas...   
2  B07XSCCZYG  Platinum Elite Softside Expandable Checked Lug...   
3  B08MVFKGJM  Freeform Hardside Expandable with Double Spinn...   
4  B01DJLKZBA  Winfield 2 Hardside Expandable Luggage with Sp...   

                                              imgUrl  \
0  https://m.media-amazon.com/images/I/815dLQKYIY...   
1  https://m.media-amazon.com/images/I/81bQlm7vf6...   
2  https://m.media-amazon.com/images/I/71EA35zvJB...   
3  https://m.media-amazon.com/images/I/91k6NYLQyI...   
4  https://m.media-amazon.com/images/I/61NJoaZcP9...   

                             productURL  stars  reviews   price  listPrice  \
0  https://www.amazon.com/dp/B014TMV5YE    4.5        0  139.99       0.00   
1  https://www.amazon.com/dp/B07GDLCQXV    4.5        0  169.99     209.99   
2  https://www.amazon.com/dp/B07XSCCZYG    4.6        0  365.49     429.99   
3  https://www.amazon.com/dp/B08MVFKGJM    4.6        0  291.59     354.37   
4  https://www.amazon.com/dp/B01DJLKZBA    4.5        0  174.99     309.99   

   category_id  isBestSeller  boughtInLastMonth   id category_name  
0          104         False               2000  104     Suitcases  
1          104         False               1000  104     Suitcases  
2          104         False                300  104     Suitcases  
3          104         False                400  104     Suitcases  
4          104         False                400  104     Suitcases

In [6]:
df = df.drop(columns = ['imgUrl','productURL','id'], axis = 1) 
df.head(2)

asin                                              title  stars  \
0  B014TMV5YE  Sion Softside Expandable Roller Luggage, Black...    4.5   
1  B07GDLCQXV  Luggage Sets Expandable PC+ABS Durable Suitcas...    4.5   

   reviews   price  listPrice  category_id  isBestSeller  boughtInLastMonth  \
0        0  139.99       0.00          104         False               2000   
1        0  169.99     209.99          104         False               1000   

  category_name  
0     Suitcases  
1     Suitcases

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1426337 entries, 0 to 1426336
Data columns (total 10 columns):
 #   Column             Non-Null Count    Dtype  
---  ------             --------------    -----  
 0   asin               1426337 non-null  object 
 1   title              1426336 non-null  object 
 2   stars              1426337 non-null  float64
 3   reviews            1426337 non-null  int64  
 4   price              1426337 non-null  float64
 5   listPrice          1426337 non-null  float64
 6   category_id        1426337 non-null  int64  
 7   isBestSeller       1426337 non-null  bool   
 8   boughtInLastMonth  1426337 non-null  int64  
 9   category_name      1426337 non-null  object 
dtypes: bool(1), float64(3), int64(3), object(3)
memory usage: 110.2+ MB


In [8]:
df.describe()

stars       reviews         price     listPrice   category_id  \
count  1.426337e+06  1.426337e+06  1.426337e+06  1.426337e+06  1.426337e+06   
mean   3.999512e+00  1.807508e+02  4.337540e+01  1.244916e+01  1.237409e+02   
std    1.344292e+00  1.761453e+03  1.302893e+02  4.611198e+01  7.311273e+01   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  1.000000e+00   
25%    4.100000e+00  0.000000e+00  1.199000e+01  0.000000e+00  6.500000e+01   
50%    4.400000e+00  0.000000e+00  1.995000e+01  0.000000e+00  1.200000e+02   
75%    4.600000e+00  0.000000e+00  3.599000e+01  0.000000e+00  1.760000e+02   
max    5.000000e+00  3.465630e+05  1.973181e+04  9.999900e+02  2.700000e+02   

       boughtInLastMonth  
count       1.426337e+06  
mean        1.419823e+02  
std         8.362720e+02  
min         0.000000e+00  
25%         0.000000e+00  
50%         0.000000e+00  
75%         5.000000e+01  
max         1.000000e+05

In [9]:
df.isnull().sum()

asin                 0
title                1
stars                0
reviews              0
price                0
listPrice            0
category_id          0
isBestSeller         0
boughtInLastMonth    0
category_name        0
dtype: int64

In [10]:
df.shape

(1426337, 10)

In [11]:
df.dtypes

asin                  object
title                 object
stars                float64
reviews                int64
price                float64
listPrice            float64
category_id            int64
isBestSeller            bool
boughtInLastMonth      int64
category_name         object
dtype: object

In [12]:
avg_price = df.groupby('category_name')['price'].mean()
avg_price

category_name
Abrasive & Finishing Products                      24.389736
Accessories & Supplies                             40.378400
Additive Manufacturing Products                    53.659274
Arts & Crafts Supplies                             13.458120
Arts, Crafts & Sewing Storage                      20.637391
                                                     ...    
Women's Watches                                    81.703771
Xbox 360 Games, Consoles & Accessories             29.766046
Xbox One Games, Consoles & Accessories             29.994712
Xbox Series X & S Consoles, Games & Accessories    25.657256
eBook Readers & Accessories                        34.766684
Name: price, Length: 248, dtype: float64

In [13]:
avg_rating = df.groupby('category_name')['stars'].mean()
avg_rating

category_name
Abrasive & Finishing Products                      3.489868
Accessories & Supplies                             4.419593
Additive Manufacturing Products                    3.105893
Arts & Crafts Supplies                             4.221737
Arts, Crafts & Sewing Storage                      3.152147
                                                     ...   
Women's Watches                                    3.695628
Xbox 360 Games, Consoles & Accessories             3.683697
Xbox One Games, Consoles & Accessories             3.885092
Xbox Series X & S Consoles, Games & Accessories    1.934880
eBook Readers & Accessories                        3.924914
Name: stars, Length: 248, dtype: float64

In [14]:
df.nunique()

asin                 1426337
title                1385430
stars                     41
reviews                11861
price                  29961
listPrice              14518
category_id              248
isBestSeller               2
boughtInLastMonth         30
category_name            248
dtype: int64

## Top 5 Expensive Product

In [15]:
expensive_product = df.sort_values(by = 'price', ascending = False)
expensive_product.head()

asin                                              title  stars  \
594853  B077BCV1JT   Overland Storage Neoxl80 Storagelibrary Lto8 SAS    0.0   
108907  B0CJHXQS8R                                   replicas Marinas    0.0   
994047  B0000516QJ  CISCO Systems 1 Port ATM Enhanced Oc12/Stm4 Si...    0.0   
994048  B0000516Q1  Cisco Systems 7140 Router Dual 10/100 Fe Dual ...    0.0   
868945  B08XMXFW1Q  PARTNERS BRAND Corrugated Trash Can Plain - 40...    0.0   

        reviews     price  listPrice  category_id  isBestSeller  \
594853        0  19731.81        0.0           55         False   
108907        0  19400.00        0.0            2         False   
994047        0  16468.70        0.0           54         False   
994048        0  12629.66        0.0           54         False   
868945        0  12519.96        0.0          160         False   

        boughtInLastMonth                  category_name  
594853                  0                   Data Storage  
108907                  0              Fabric Decorating  
994047                  0               Computer Servers  
994048                  0               Computer Servers  
868945                  0  Packaging & Shipping Supplies

In [16]:
lowest_rating = df.sort_values(by = 'stars')
lowest_rating.head(4)

asin                                              title  stars  \
611826  B0C687K1HQ  Marquee Light Bulbs 13.8 Ft Long String Lights...    0.0   
702986  B0C7H1BRVN  SWAMISP Set of 1 India Rakhi for Brothers Desi...    0.0   
702984  B0BSGT4NL6      Classics Goldtone Hammered Disk Drop Earrings    0.0   
702983  B0C26TK38W  Scottish Thistle Gifts Sassenach Jewelry Outla...    0.0   

        reviews  price  listPrice  category_id  isBestSeller  \
611826        0  34.99        0.0          168         False   
702986        0   4.99        0.0           88         False   
702984        0   8.99        0.0           88         False   
702983        0  29.88        0.0           88         False   

        boughtInLastMonth   category_name  
611826                  0  Seasonal Décor  
702986                  0   Boys' Jewelry  
702984                  0   Boys' Jewelry  
702983                  0   Boys' Jewelry

## Total Product in each Categories 

In [29]:
total_category_count = df.groupby('category_name').asin.count()
total_category_count

category_name
Abrasive & Finishing Products                      8646
Accessories & Supplies                             4226
Additive Manufacturing Products                    7619
Arts & Crafts Supplies                             8313
Arts, Crafts & Sewing Storage                      7592
                                                   ... 
Women's Watches                                    6542
Xbox 360 Games, Consoles & Accessories             3809
Xbox One Games, Consoles & Accessories             3582
Xbox Series X & S Consoles, Games & Accessories    5645
eBook Readers & Accessories                         582
Name: asin, Length: 248, dtype: int64

## Total Best Seller 

In [30]:
df.groupby('isBestSeller').asin.count()

isBestSeller
False    1417817
True        8520
Name: asin, dtype: int64

There are only 8520 Best seller in this dataset.